### PR TITLE
plan: combine agg_prune and agg_pushdown.

### PR DESCRIPTION
**Pull request by hanfei1991**
after agg push down, we can always prune the old aggregation.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: plan/aggregation_push_down.go [hunk 1]
```diff
@@ -315,19 +315,23 @@ func (a *aggPushDownSolver) pushAggCrossUnion(agg *Aggregation, unionSchema *exp
...
 	return p, nil
 }
 
 // aggPushDown tries to push down aggregate functions to join paths.
-func (a *aggPushDownSolver) aggPushDown(p LogicalPlan) {
+func (a *aggregationOptimizer) aggPushDown(p LogicalPlan) LogicalPlan {
 	if agg, ok := p.(*Aggregation); ok {
+		proj := a.tryToEliminateAggregation(agg)
+		if proj != nil {
+			return proj
```
Comment (winoros) [inline plan/aggregation_push_down.go]: sql like `select count(1) from (select count(1), a, b from t group by a, b) tt group by b;`, if both a and b are key, the inner aggregation won't be eliminated.

Comment (hanfei1991) [inline plan/aggregation_push_down.go]: well, you are right!

**Comment by hanfei1991** [comment]
@winoros PTAL

**Comment by winoros** [comment]
LGTM

**Pull request merged**